proof_of_stake/src/types.rs

```diff
@@ -40,6 +40,13 @@ pub type ValidatorConsensusKeys = crate::epoched::Epoched<
     crate::epoched::OffsetDefaultNumPastEpochs,
 >;
 
+/// Epoched validator's protocol key.
+pub type ValidatorProtocolKeys = crate::epoched::Epoched<
+    common::PublicKey,
+    crate::epoched::OffsetPipelineLen,
+    crate::epoched::OffsetMaxProposalPeriodPlus,
```

Comment:
I think the offset can be reduced in the future. I'm not sure what the appropriate value for this should be, though.